Automattic/jetpack
# Fix version conflicts for @types/react

Due to type conflict between v17 and v18 of `@types/react`, we have some TypeScript errors in certain components. For example on `master` branch, if you run `cd projects/js-packages/components/ && pnpm tsc`, you will see TS error(s).

#### Changes proposed in this Pull Request:

* Override @types/react^17 peer dependency for @testing-library/react-hooks in order to use @types/react^18
* Upgrade `@types/react` to v18

#### Jetpack product discussion
See https://github.com/Automattic/jetpack/pull/24256#discussion_r866101448  and https://github.com/Automattic/jetpack/pull/24276#discussion_r867823262

#### Does this pull request change what data or activity we track or use?
No

#### Testing instructions:

* Checkout the PR
* Run `jetpack install --all` or `pnpm install`
* Confirm that you don't seen any MISSING PEER DEPENDENCY error.
* Run `cd projects/js-packages/components/ && pnpm tsc`
* Confirm that you don't see any TS error


#### Next Steps
I think it's the time to make TS check a part of the build process by simply running the TypeScript compiler CLI - `tsc` before running the actual build commands to avoid any errors being left in TS components.
It will be as simple as adding a `"build": "tsc && ..."` script to `js-packages/components` and `plugins/jetpack` etc.
Thoughts @Automattic/jetpack-agora ?
